components/NumberPicker/Slider.tsx

```diff
@@ -0,0 +1,132 @@
+import css from "@styled-system/css";
+import styled from "styled-components";
+import { throttle } from "throttle-debounce";
+import { useCallback, ChangeEvent, ReactNode, useState } from "react";
+import Flex, { FlexProps } from "components/Flex";
+import { all, StyledSystemProps, transition } from "components/system";
+
+export interface SliderProps {
+  onChange: (val: number) => void;
+  maxValue?: number;
+  minValue?: number;
+  initialValue?: number;
+  label?: string;
+  changeThrottle?: number;
+  renderMin?: (val: number) => ReactNode;
+  renderMax?: (val: number) => ReactNode;
+  renderValue?: (val: number) => ReactNode;
+}
+
+const DEFAULT_CHANGE_THROTTLE = 100;
+
+const echo = <T extends unknown>(thing: T): T => thing;
+
+export function Slider({
+  onChange,
+  label,
+  maxValue = 50,
+  minValue = 1,
+  changeThrottle = DEFAULT_CHANGE_THROTTLE,
+  initialValue: value = minValue,
+  renderMin = echo,
+  renderMax = echo,
+  renderValue = echo,
+  ...props
+}: SliderProps & FlexProps) {
+  const limitedValue = Math.min(maxValue, Math.max(minValue, value));
+  const [innerValue, setInnerValue] = useState(limitedValue);
+  const throttledOnChange = useCallback(throttle(changeThrottle, onChange), [
+    changeThrottle,
+    onChange,
+  ]);
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const value = Number(e.target.value);
+      setInnerValue(value);
+      throttledOnChange(value);
+    },
+    [onChange]
+  );
+
+  return (
+    <Flex
+      display="inline-flex"
+      alignItems={label ? "center" : "baseline"}
+      {...props}
+    >
+      <StyledSpan>{renderMin(minValue)}</StyledSpan>
+      <Flex flexDirection="column" alignItems="center" width="100%" px="2">
+        {label && <StyledSpan>{label}</StyledSpan>}
+        <StyledRange
+          type="range"
+          onChange={handleChange}
+          value={innerValue}
+          min={minValue}
+          max={maxValue}
+        />
+        <StyledSpan color="light-purple" fontWeight="bold">
+          {renderValue(innerValue)}
+        </StyledSpan>
+      </Flex>
+      <StyledSpan>{renderMax(maxValue)}</StyledSpan>
+    </Flex>
+  );
+}
+
+const StyledSpan = styled("span")<StyledSystemProps>(
+  css({
+    color: "gray",
+    fontSize: "text-sm",
+    lineHeight: "12px",
+  }),
+  all
+);
+
+const thumbStyles = css({
+  appearance: "none",
+  backgroundColor: "white",
+  border: "1px solid",
+  borderColor: "light-gray",
+  borderRadius: "circle",
+  width: 22,
+  height: 22,
+  boxShadow: "0 0 4px #D2DBDF",
+  transition: transition([["backgroundColor", "interaction"]]),
+  cursor: "ew-resize",
+});
+
+const StyledRange = styled("input")<StyledSystemProps>(
+  css({
+    width: "100%",
+    height: 10,
```

Comment:
```suggestion
    height: "10px",
```

I think we will add 10th element to the theme sooner or later.

Reply:
After some experiments I realized that `height` and `width` doesn't resolve `space` from styled system. ¯\_(ツ)_/¯
It just translates number to pixels value